pkg/controller/nodelifecycle/scheduler/taint_manager.go

```diff
@@ -115,6 +119,27 @@ func deletePodHandler(c clientset.Interface, emitEventFunc func(types.Namespaced
 	}
 }
 
+func addConditionAndDeletePod(ctx context.Context, c clientset.Interface, name, ns string) (err error) {
+	if feature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
+		pod, err := c.CoreV1().Pods(ns).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
```

Comment:
you return error if is a "NotFound" errror because the pod was already deleted, or that can not happen?

Reply:
It may happen, if the pods is deleted since queuing for eviction by the taint manager: https://github.com/kubernetes/kubernetes/blob/9f9602ef830d55e844ad500c0a3ab342dd7cc98d/pkg/controller/nodelifecycle/scheduler/taint_manager.go#L382
however, the already called Delete method would also return NotFound error so it seems consistent to also return NotFound from the get.

Reply:
+1 if the error is NotFound error, we should return success and bail out

Reply:
ignore that, the default behavior is good.